geemap/basemaps.py

```diff
@@ -23,36 +23,61 @@
 import folium
 import ipyleaflet
 import xyzservices
-from .common import check_package, planet_tiles
-
-XYZ_TILES = {
-    "OpenStreetMap": {
-        "url": "https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png",
-        "attribution": "OpenStreetMap",
-        "name": "OpenStreetMap",
-    },
-    "ROADMAP": {
-        "url": "https://server.arcgisonline.com/ArcGIS/rest/services/World_Street_Map/MapServer/tile/{z}/{y}/{x}",
-        "attribution": "Esri",
-        "name": "Esri.WorldStreetMap",
-    },
-    "SATELLITE": {
-        "url": "https://server.arcgisonline.com/ArcGIS/rest/services/World_Imagery/MapServer/tile/{z}/{y}/{x}",
-        "attribution": "Esri",
-        "name": "Esri.WorldImagery",
-    },
-    "TERRAIN": {
-        "url": "https://server.arcgisonline.com/ArcGIS/rest/services/World_Topo_Map/MapServer/tile/{z}/{y}/{x}",
-        "attribution": "Esri",
-        "name": "Esri.WorldTopoMap",
-    },
-    "HYBRID": {
-        "url": "https://server.arcgisonline.com/ArcGIS/rest/services/World_Imagery/MapServer/tile/{z}/{y}/{x}",
-        "attribution": "Esri",
-        "name": "Esri.WorldImagery",
-    },
-}
-
+from .common import (
+    check_package,
+    planet_tiles,
+    google_map_tiles,
+    google_maps_api_key,
+)
+
+MAPS_API_KEY = google_maps_api_key()
```

Comment:
We should avoid global variables. Instead of determining whether a Maps API key is set on module initialization, let's do it when a Map instance is created.

If users `import geemap` before setting the environment variable or Colab secret, users won't get Google Maps baselayers.

Reply:
Global variable `MAPS_API_KEY` has been removed.